packages/plugin-data/src/dataCommand.ts

```diff
@@ -169,13 +169,19 @@ export abstract class DataCommand extends SfdxCommand {
         throw new Error(messages.getMessage('TextUtilMalformedKeyValuePair', [pair]));
       } else {
         const key = pair.substr(0, eqPosition);
-        constructedObject[key] = pair.substr(eqPosition + 1);
+        constructedObject[key] = this.convertToBooleanIfApplicable(pair.substr(eqPosition + 1));
       }
     });
 
     return constructedObject;
   }
 
+  private convertToBooleanIfApplicable(input: string): boolean | string {
+    if (input.toLowerCase() === 'false') return false;
```

Comment:
@mshanemc You may want to trim the input. Not sure how strict key/value pairs are treated on the command line. 
```
sfdx force:data:record:get -s Account -w "Name='Universal Containers' Foo=true"
sfdx force:data:record:get -s Account -w "Name='Universal Containers' Foo= true"
```

Reply:
good point.  I put it in `transformKeyValueSequence`

Reply:
@mshanemc Not sure that is the right place. What if there are intentional leading or trailing whitespace is string values?